test/account_api_test.rb

```diff
@@ -1,81 +1,407 @@
 require "test_helper"
 require "gds_api/account_api"
-require "gds_api/test_helpers/account_api"
 
 describe GdsApi::AccountApi do
-  include GdsApi::TestHelpers::AccountApi
+  include PactTest
 
-  let(:base_url)      { Plek.find("account-api") }
-  let(:api_client)    { GdsApi::AccountApi.new(base_url) }
+  let(:api_client) { GdsApi::AccountApi.new(account_api_host) }
 
-  it "gets a sign in URL" do
-    stub_account_api_get_sign_in_url(auth_uri: "https://www.example.com")
-    assert_equal("https://www.example.com", api_client.get_sign_in_url.to_hash["auth_uri"])
+  let(:authenticated_headers) { { "GOVUK-Account-Session" => govuk_account_session } }
+  let(:govuk_account_session) { "logged-in-user-session" }
+
+  describe "getting a sign-in URL" do
+    let(:redirect_path) { nil }
+
+    before do
+      account_api
+        .given("no particular state")
+        .upon_receiving("a sign-in request")
+        .with(
+          method: :get,
+          path: "/api/oauth2/sign-in",
+          headers: GdsApi::JsonClient.default_request_headers,
+        )
+        .will_respond_with(
+          status: 200,
+          headers: { "Content-Type" => "application/json; charset=utf-8" },
+          body: {
+            auth_uri: Pact.like("string"),
+            state: Pact.like("string"),
+          },
+        )
+    end
+
+    it "responds with 200 OK, an authentication URI, and a state for CSRF protection" do
+      response = api_client.get_sign_in_url
+      assert response["auth_uri"].present?
+      assert response["state"].present?
+      assert_equal 200, response.code
+    end
   end
 
-  it "gives a session ID if the auth response validates" do
-    stub_account_api_validates_auth_response(code: "foo", state: "bar")
-    assert(!api_client.validate_auth_response(code: "foo", state: "bar")["govuk_account_session"].nil?)
+  describe "getting a sign-in URL, with a redirect_path pointing to another domain" do
+    let(:redirect_path) { "//protocol-relative-redirect" }
+
+    before do
+      account_api
+        .given("no particular state")
+        .upon_receiving("a sign-in request with an invalid redirect_path")
+        .with(
+          method: :get,
+          path: "/api/oauth2/sign-in",
+          headers: GdsApi::JsonClient.default_request_headers,
+          query: { redirect_path: Pact.like(redirect_path) },
+        )
+        .will_respond_with(
+          status: 422,
+          headers: { "Content-Type" => "application/json; charset=UTF-8" },
+        )
+    end
+
+    it "responds with 422 Unprocessable Entity" do
+      assert_raises GdsApi::HTTPUnprocessableEntity do
+        api_client.get_sign_in_url(redirect_path: redirect_path)
+      end
```

Comment:
A quick observation of mine is that we probably don't need pact tests for negative scenarios when this app doesn't have any code to handle those scenarios. It doesn't seem that app needs a contract that Account API will act this way in a particular scenario because it doesn't have a view on it.

(I imagine this is a carryover from Publishing API ones that may have a bunch of scenarios tested that may not actually add value).

The positive ones I've glanced at look good though. I've not got time for a full review but awesome to see you're working on these and seeking feedback 👍

Reply:
Sure, I see where you're coming from. And reading [the pact docs](https://docs.pact.io/consumer/), it says you shouldn't use pact tests to ensure functional correctness of the provider.  So I'll remove the error case tests.